Managers/UInputMan.cpp

```diff
@@ -1,2858 +1,1393 @@
+			}
 
+			// Determine deadzone settings
+			float deadZone = 0.0F;
+			int deadZoneType = DeadZoneType::CIRCLE;
+			int player = -1;
 
-bool UInputMan::AccumulatedElementReleased(int element)
-{
-	if (element < 0 || element >= INPUT_COUNT)
-		return false;
+			// Retrieve deadzone settings from player's input scheme
+			for (int p = 0; p < UInputMan::MAX_PLAYERS; p++) {
+				int device = m_ControlScheme[p].GetDevice() - DEVICE_GAMEPAD_1;
+
+				if (device == joystick) {
+					deadZone = m_ControlScheme[p].GetJoystickDeadzone();
+					deadZoneType = m_ControlScheme[p].GetJoystickDeadzoneType();
+					player = p;
+					break;
+				}
+			}
+
+			// Disable input if it's in circle deadzone
+			if (player > -1 && deadZoneType == DeadZoneType::CIRCLE && deadZone > 0.0F) {
+				const InputMapping *element = m_ControlScheme[player].GetInputMappings();
+				Vector aimValues;
+
+				// Left stick
+				if (element[INPUT_L_LEFT].JoyDirMapped()) { aimValues.m_X = AnalogAxisValue(joystick, element[INPUT_L_LEFT].GetStick(), element[INPUT_L_LEFT].GetAxis()); }
+				if (element[INPUT_L_UP].JoyDirMapped()) { aimValues.m_Y = AnalogAxisValue(joystick, element[INPUT_L_UP].GetStick(), element[INPUT_L_UP].GetAxis()); }
+
+				if (aimValues.GetMagnitude() < deadZone * 2) {
+					if (element[INPUT_L_LEFT].JoyDirMapped()) {
+						if (joy[joystick].stick[element[INPUT_L_LEFT].GetStick()].flags & JOYFLAG_UNSIGNED) {
+							joy[joystick].stick[element[INPUT_L_LEFT].GetStick()].axis[element[INPUT_L_LEFT].GetAxis()].pos = 128;
+							joy[joystick].stick[element[INPUT_L_LEFT].GetStick()].axis[element[INPUT_L_LEFT].GetAxis()].d1 = 0;
+							joy[joystick].stick[element[INPUT_L_LEFT].GetStick()].axis[element[INPUT_L_LEFT].GetAxis()].d2 = 0;
+						} else {
+							joy[joystick].stick[element[INPUT_L_LEFT].GetStick()].axis[element[INPUT_L_LEFT].GetAxis()].pos = 0;
+							joy[joystick].stick[element[INPUT_L_LEFT].GetStick()].axis[element[INPUT_L_LEFT].GetAxis()].d1 = 0;
+							joy[joystick].stick[element[INPUT_L_LEFT].GetStick()].axis[element[INPUT_L_LEFT].GetAxis()].d2 = 0;
+						}
+					}
+					if (element[INPUT_L_UP].JoyDirMapped()) {
+						if (joy[joystick].stick[element[INPUT_L_UP].GetStick()].flags & JOYFLAG_UNSIGNED) {
+							joy[joystick].stick[element[INPUT_L_UP].GetStick()].axis[element[INPUT_L_UP].GetAxis()].pos = 128;
+							joy[joystick].stick[element[INPUT_L_UP].GetStick()].axis[element[INPUT_L_UP].GetAxis()].d1 = 0;
+							joy[joystick].stick[element[INPUT_L_UP].GetStick()].axis[element[INPUT_L_UP].GetAxis()].d2 = 0;
+						} else {
+							joy[joystick].stick[element[INPUT_L_UP].GetStick()].axis[element[INPUT_L_UP].GetAxis()].pos = 0;
+							joy[joystick].stick[element[INPUT_L_UP].GetStick()].axis[element[INPUT_L_UP].GetAxis()].d1 = 0;
+							joy[joystick].stick[element[INPUT_L_UP].GetStick()].axis[element[INPUT_L_UP].GetAxis()].d2 = 0;
+						}
+					}
+				}
+
+				aimValues.Reset();
```

Comment:
I guess this is done before looking at right stick cause that's for aiming, and we wanna make sure the vector is totally clean before doing stuff with it?

Reply:
Yeah.